src/ui/flutter_app/lib/widgets/game_page.dart

```diff
@@ -1585,147 +1607,107 @@ class _GamePageState extends State<GamePage>
       them = S.of(context).player1;
     }
 
-    String tip = (_tip == null || !Config.screenReaderSupport) ? "" : "\n$_tip";
+    final String tip =
+        (_tip == null || !Config.screenReaderSupport) ? "" : "\n$_tip";
 
     String lastMove = "";
-    if (pos.recorder != null &&
-        pos.recorder.lastMove != null &&
-        pos.recorder.lastMove.notation != null) {
-      String n1 = pos.recorder.lastMove.notation;
+    if (pos.recorder?.lastMove?.notation != null) {
+      final String n1 = pos.recorder!.lastMove!.notation!;
 
       if (n1.startsWith("x")) {
-        String n2 = pos.recorder.moveAt(pos.recorder.movesCount - 2).notation;
+        final String n2 =
+            pos.recorder!.moveAt(pos.recorder!.movesCount - 2).notation!;
         lastMove = n2 + n1;
       } else {
         lastMove = n1;
       }
       if (Config.screenReaderSupport) {
-        lastMove = S.of(context).lastMove +
-            ": " +
-            them +
-            ", " +
-            lastMove +
-            period +
-            "\n";
+        lastMove = "${S.of(context).lastMove}: $them, $lastMove$period\n";
       } else {
-        lastMove = S.of(context).lastMove + ": " + lastMove + period + "\n";
+        lastMove = "${S.of(context).lastMove}: $lastMove$period\n";
       }
     }
 
     String addedPeriod = "";
     if (Config.screenReaderSupport &&
-        tip.length >= 1 &&
+        tip.isNotEmpty &&
         tip[tip.length - 1] != '.' &&
         tip[tip.length - 1] != '!') {
       addedPeriod = ".";
     }
 
-    String ret = phase +
-        period +
-        "\n" +
-        lastMove +
-        S.of(context).sideToMove +
-        ": " +
-        us +
-        period +
-        tip +
-        addedPeriod +
-        "\n\n" +
-        S.of(context).pieceCount +
-        ":\n" +
-        pieceCountInHand +
-        S.of(context).player1 +
-        " " +
-        S.of(context).onBoard +
-        ": " +
-        pos.pieceOnBoardCount[PieceColor.white].toString() +
-        comma +
-        "\n" +
-        S.of(context).player2 +
-        " " +
-        S.of(context).onBoard +
-        ": " +
-        pos.pieceOnBoardCount[PieceColor.black].toString() +
-        period +
-        "\n\n" +
-        S.of(context).score +
-        ":\n" +
-        S.of(context).player1 +
-        ": " +
-        pos.score[PieceColor.white].toString() +
-        comma +
-        "\n" +
-        S.of(context).player2 +
-        ": " +
-        pos.score[PieceColor.black].toString() +
-        comma +
-        "\n" +
-        S.of(context).draw +
-        ": " +
-        pos.score[PieceColor.draw].toString() +
-        period;
+    final String ret =
```

Comment:
It's incredible how long a line can be here without a line break. I'm not sure if my Android Studio and VSCode can read the relevant configuration in the workspace to make sure it doesn't format this code back into the original format.